src/main/scala/tech/cryptonomic/conseil/Conseil.scala

```diff
@@ -31,115 +42,157 @@ object Conseil
     with ConseilOutput {
 
   loadApplicationConfiguration(args) match {
-    case Right((server, platforms, securityApi, verbose, metadataOverrides, nautilusCloud)) =>
+    case Left(errors) =>
+    //nothing to do
+    case Right((server, platforms, securityApi, failFast, verbose, metadataOverrides, nautilusCloud)) =>
       implicit val system: ActorSystem = ActorSystem("conseil-system")
       implicit val materializer: ActorMaterializer = ActorMaterializer()
       implicit val executionContext: ExecutionContextExecutor = system.dispatcher
-      nautilusCloud.foreach { ncc =>
-        system.scheduler.schedule(ncc.delay, ncc.interval)(Security.updateKeys(ncc))
-      }
 
-      val validateApiKey: Directive[Tuple1[String]] = optionalHeaderValueByName("apikey").tflatMap[Tuple1[String]] {
-        apiKeyTuple =>
-          val apiKey = apiKeyTuple match {
-            case Tuple1(apiKey) => apiKey
-            case _ => None
-          }
+      val retries = if (failFast.on) Some(0) else None
+
+      val serverBinding =
+        retry(maxRetry = retries, deadline = Some(server.startupDeadline fromNow))(
+          initServices(server, platforms, metadataOverrides, nautilusCloud).get
```

Comment:
How come we `get` from the Try returned by `initServices()` and not explicitly handle any errors?

Reply:
That's how the `retry` logic expects to work.
It spawns a "thunk" of code in a Future, and internally catches any error to be able to handle the retry logic independently.

It provides parameters to define how long to retry, how long to wait for each retry and if there's any specific expection to pass through without retrying, and simply failing the call immediately